lang/test/org/partiql/lang/eval/evaluatortestframework/AbstractPipeline.kt

```diff
@@ -0,0 +1,15 @@
+package org.partiql.lang.eval.evaluatortestframework
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.TypingMode
+
+/**
+ * Represents an abstract pipeline (either [org.partiql.lang.CompilerPipeline] or
+ * [org.partiql.lang.planner.PlannerPipeline]) so that [PipelineEvaluatorTestAdapter] can work with either.
+ *
+ * Includes only those properties and methods that are required for testing purposes.
+ */
+interface AbstractPipeline {
```

Comment:
Could we apply the `AbstractPipeline` outside the test class (i.e. have `CompilerPipeline` and `PlannerPipeline` implement this interface)? Otherwise, I would try to make it more clear from the interface name that this relates to just testing.

Reply:
I will consider this.

Reply:
So... I don't think having a `typingMode` directly on the `CompilerPipeline` (or `PlannerPipeline` introduced in #590) is the right API because this is actually a property of `CompileOptions` (and `EvaluatorOptions` also introduced in introduced in #590).  This is actually the reason I didn't do what you suggest here already.  I have some additional thoughts about the API of `EvaluatorOptions` / `PlannerPipeline`, which I  think are best [addressed separately](https://github.com/partiql/partiql-lang-kotlin/issues/598).